.github/workflows/integration_test.yml

```diff
@@ -10,13 +10,13 @@ jobs:
       matrix:
         kube_provider: [kind]
     steps:
-      - uses: actions/checkout@main
-      - uses: actions/setup-go@main
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3
+      - uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 # v4
```

Comment:
May I ask why are you using commit hash for the action instead of the version, like: 

```yaml
  - uses: actions/checkout@v3
  - uses: actions/setup-go@v4
```

Reply:
It is a security best practice for open source projects that people use you can google the reason

Reply:
an example blog here: https://michaelheap.com/ensure-github-actions-pinned-sha/

Reply:
this article is not a best practice. It's just some points of view. None of the reference link in this article from github action official. Hence please rollback those lines. It just try to sales some 3rd party github action or personal point.